src/views/connection.vue

```diff
@@ -9,41 +9,136 @@ export default {
     }
   },
   computed: {
-    ...mapState(['collectionName', 'databaseName', 'records', 'totalNumberOfResults'])
+    ...mapState([
+      'openedCollections',
+      'collectionName',
+      'databaseName',
+      'activeRecords',
+      'collectionRecords',
+      'totalNumberOfResults'
+    ])
+  },
+  watch: {
+    collectionName: function(newValue) {
+      if (this.collectionRecords[newValue]) {
+        this.queryJson = this.collectionRecords[newValue].query
+      } else {
+        this.queryJson = '{}'
+      }
+    }
   },
   methods: {
-    ...mapActions(['find']),
+    ...mapActions(['find', 'setCollection', 'removeCollectionFromOpenedCollections']),
     async doQuery() {
+      if (!this.queryJson) {
+        this.queryJson = '{}'
+      }
       await this.find(JSON.parse(this.queryJson))
     }
   }
 }
 </script>
 <template lang="pug">
-  div.connection
-    h1 {{ databaseName }} - {{ collectionName }} collection
-    .box
+.connections(v-if='databaseName')
```

Comment:
this view should always be visible, otherwise the app looks really odd. but we can have a welcome message to guide user to connect to a database first.